src/main/java/org/reactome/addlinks/referencecreators/EnsemblBioMartMicroarrayPopulator.java

```diff
@@ -0,0 +1,159 @@
+        for (String protein : rgpProteins) {
+            if (EnsemblBioMartUtil.getProteinToTranscriptsMappings(speciesBiomartName, mappings).containsKey(protein)) {
+                for (String transcript : EnsemblBioMartUtil.getProteinToTranscriptsMappings(speciesBiomartName, mappings).get(protein)) {
+                    if (EnsemblBioMartUtil.getTranscriptToMicroarraysMappings(speciesBiomartName, mappings).containsKey(transcript)) {
+                        for (String microarrayId : EnsemblBioMartUtil.getTranscriptToMicroarraysMappings(speciesBiomartName, mappings).get(transcript)) {
+                            Collection<String> otherIdentifiers = rgpInst.getAttributeValuesList(ReactomeJavaConstants.otherIdentifier);
+                            if (!otherIdentifiers.contains(microarrayId) && !this.testMode) {
+                                rgpInst.addAttributeValue(ReactomeJavaConstants.otherIdentifier, microarrayId);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        sortOtherIdentifiersAndUpdateInstance(rgpInst);
+    }
+
+    /**
+     * Find proteins associated with ReferenceGeneProduct instance.
+     * @param rgpIdentifier - String,Identifier associated with RGP instance.
+     * @param rgpInst - GKInstance, ReferenceGeneProduct instance
+     * @param speciesBiomartName - String, Species name in BioMart format (eg: hsapiens).
+     * @param mappings - Map<String, Map< String, List<String>>>, Super mapping structure generated by EnsemblBioMartFileProcessor.
+     * @return - Set<String>, Set of proteins associated with RGP instance.
+     * @throws Exception - Thrown when retrieving data from GKInstance that does not exist or invalid for the GKInstance class.
+     */
+    public Set<String> findRGPProteins(String rgpIdentifier, GKInstance rgpInst, String speciesBiomartName, Map<String, Map<String, List<String>>> mappings) throws Exception {
+        Set<String> rgpProteins = new HashSet<>();
+        GKInstance refDbInst = (GKInstance) rgpInst.getAttributeValue(ReactomeJavaConstants.referenceDatabase);
+        // If reference DB is Ensembl, add the identifier associated with RGP instance.
+        if (refDbInst.getDisplayName().toLowerCase().contains("ensembl")) {
+            if (EnsemblBioMartUtil.getProteinToTranscriptsMappings(speciesBiomartName, mappings).containsKey(rgpIdentifier)) {
+                rgpProteins.add(rgpIdentifier);
+            }
+        // If reference DB is Uniprot, get Ensembl protein identifier associated with
+        // instance's identifier attribute, which should be a Uniprot identifier.
+        } else if (refDbInst.getDisplayName().toLowerCase().contains("uniprot")) {
+            if (EnsemblBioMartUtil.getUniprotToProteinsMappings(speciesBiomartName, mappings) != null && EnsemblBioMartUtil.getUniprotToProteinsMappings(speciesBiomartName, mappings).containsKey(rgpIdentifier)) {
```

Comment:
The null check shouldn't be necessary as long as the `getUniprotToProteinsMappings` method is guaranteed to return a map using `computeIfAbsent`.

Reply:
Done